web/modules/custom/pipeline_log/pipeline_log.module

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for Pipeline logging module.
+ */
+
+declare(strict_types = 1);
+
+use Drupal\user\UserInterface;
+
+/**
+ * Implements hook_user_login().
+ */
+function pipeline_log_user_login(UserInterface $account) {
+  if (!$account->hasPermission('access pipeline selector')) {
+    // If the user does not have access to run a pipeline, they should not
+    // receive warnings about them either.
+    return;
+  }
+
+  $time = \Drupal::getContainer()->get('datetime.time');
+  $pipeline_manager = \Drupal::getContainer()->get('plugin.manager.pipeline_pipeline');
+  // 90 days of interval.
+  $interval = 60 * 60 * 24 * 30 * 3;
+  foreach (\Drupal::getContainer()->get('keyvalue')->get('pipeline_log')->getAll() as $pipeline_id => $last_execute) {
+    if (!$account->hasPermission("execute {$pipeline_id} pipeline")) {
+      // Only show information about pipelines they have the ability to execute.
+      continue;
+    }
+
+    $time_since_last_run = $time->getRequestTime() - $last_execute;
+    if ($time_since_last_run >= $interval) {
+      $pipeline = $pipeline_manager->getDefinition($pipeline_id);
```

Comment:
What if the pipeline has been removed in the meantime? I know us less expected but it's cheap to add an `if(...) {...}` and it won't hurt.